src/MTGCore.ClientApp/ClientApp/src/App.vue

```diff
@@ -44,138 +45,156 @@
-import "bootstrap/dist/css/bootstrap.min.css";
-import axios from "axios";
+import 'bootstrap';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import axios from 'axios';
 
 export default {
-  name: "app",
+  name: 'app',
   data: function () {
     return {
-      Name: "",
+      Name: '',
       searchedCards: null,
-      searchedCardsLoadingError: "",
+      searchedCardsLoadingError: '',
       searchInProgress: false,
       deckCards: null,
       deckList: null,
-      selectedDeck: null
+      selectedDeck: null,
+      searchFilter: null,
     };
   },
   components: {
     ResultsTable,
     DeckCardList,
-    DeckList
+    DeckList,
+    SearchFilterModal,
   },
   methods: {
+    SetSearchFilter(value) {
+      this.searchFilter = value;
+    },
+    openmodal: function () {
+      this.$refs.SearchFilterRef.open();
+    },
     addNewDeck(title) {
       var params = new URLSearchParams();
       params.append('title', title);
 
       axios({
-        method: "post",
+        method: 'post',
         url: `https://localhost:44305/api/Deck/New`,
-        data: params
+        data: params,
       })
-          .then(res => {
-            this.updateDeckCardList(res.data);
-            this.updateDeckList();
-            // Set selected dropdown value to new deck ID
-            this.selectedDeck = res.data;
-            // Set textbox val back to empty string
-            this.$refs.deckListRef.resetNewDeckName();
-          })
-          .catch(err => {
-            alert(`There was an error submitting your form. See details: ${err}`);
-          });
+        .then((res) => {
+          this.updateDeckCardList(res.data);
+          this.updateDeckList();
+          // Set selected dropdown value to new deck ID
+          this.selectedDeck = res.data;
+          // Set textbox val back to empty string
+          this.$refs.deckListRef.resetNewDeckName();
+        })
+        .catch((err) => {
+          alert(`There was an error submitting your form. See details: ${err}`);
+        });
     },
     addToDeck(id) {
       axios({
-        method: "post",
+        method: 'post',
         url: `https://localhost:44305/api/Deck/Add/${this.selectedDeck}/${id}`,
-        data: this.$data
+        data: this.$data,
       })
-          .then(() => {
-            this.updateDeckCardList(this.selectedDeck);
-          })
-          .catch(err => {
-            alert(`There was an error submitting your form. See details: ${err}`);
-          });
+        .then(() => {
+          this.updateDeckCardList(this.selectedDeck);
+        })
+        .catch((err) => {
+          alert(`There was an error submitting your form. See details: ${err}`);
+        });
     },
     SubmitForm() {
       this.searchInProgress = true;
       this.searchedCards = null;
       this.searchedCardsLoadingError = null;
+
+
+      var filter = {}
+      filter["SearchFilter"] = this.searchFilter;
+      filter["SearchFilter"].Name = this.Name;
```

Comment:
I think it would be nice to have the filter as a variable in the `data` function. Then you could just pass it into the axios function without requiring any set-up.

This would tie-in nicely with including the name in that component too as you would essentially replace the name object here with the filter object.